crates/sui-core/src/unit_tests/authority_tests.rs

```diff
@@ -1212,20 +1199,20 @@ async fn test_handle_certificate_interrupted_retry() {
             .unwrap()
             .unwrap();
 
-        let shared_object_cert = make_test_transaction(
+        // Use owned object certificate for the LimitedPoll test.
+        // Shared object certificate requires more setup before execution. And if we rely on
+        // send_consensus() for setup, the certificate will also get executed as output in
+        // handle_consensus_transaction(), making the LimitedPoll test ineffective.
```

Comment:
Sorry, but the shared object case was added very deliberately in #4579 - I think we need to leave the test as is to prevent a regression.

Reply:
Reverted the test to use certificate with shared objects. Added a function to run the preparation logic for executing certificates out of consensus, without actual execution.